### PR TITLE
Firefly-Queueserver improvements

### PR DESCRIPTION
**Pull request by canismarko**
1) Workaround that allows ``bps.mv`` to be added to the queueserver.
2) The Firefly queue client should now reconnect better (needs to be checked

Things to do before merging:

- [x] add tests
- [x] flake8, black, and isort
- [x] check at the beamline

### CODE REVIEW

**Comment by canismarko** [comment]
Tests won't pass until #320 is merged.

File: src/firefly/main_window.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 from haven import load_config
 
+from .queue_client import is_in_use
```
Comment (sterbinsky) [inline src/firefly/main_window.py:13]: Seems strange this package starts with a "."

File: src/firefly/queue_client.py [hunk 2]
```diff
@@ -25,13 +35,73 @@ def queueserver_api():
...
+    ==========
+    status_mapping
+      Maps the queuestatus parameters that are sent onto the update
+      signal names to yield
+
+    Yields
+    ======
+    to_update
+      Dictionary with signals to emit as keys, and tuples of
+      ``*args`` to emit and values. So ``{self.status_changed:
```
Comment (sterbinsky) [inline src/firefly/queue_client.py:51]: Should "and" be "as" in this sentence? It seems off gramatically

File: src/firefly/queue_client.py [hunk 3]
```diff
@@ -45,16 +115,20 @@ class QueueClient(QObject):
...
         self._last_queue_status = {}
         # Setup timer for updating the queue
         self.timer = QTimer()
         self.timer.timeout.connect(self.update)
+        # Create the generator to keep track of the queue states
+        self.status = queue_status(status_mapping=self.parameter_mapping)
+        next(self.status)  # Prime the generator
+
+    def start(self):
+        # Start the time so that it
```
Comment (sterbinsky) [inline src/firefly/queue_client.py:130]: comment appears incomplete

**Review by sterbinsky — APPROVED** [review]
Made a few in-line comments

File: src/firefly/main_window.py (hunk 1 above)
Comment (canismarko) [reply src/firefly/main_window.py:13]: That's a relative import. It's equivalent to `from firefly.queue_client import is_in_use`.

PEP8 recommends the absolute import. I'll update it. https://peps.python.org/pep-0008/

**Comment by canismarko** [comment]
Thanks @sterbinsky. I made those changes and will merge now.

**Pull request merged**